source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRInputLabelView.mm

```diff
@@ -321,41 +353,15 @@ - (void)getInput:(NSMutableDictionary *)dictionary
     }
 }
 
-+ (void)commonSetFocus:(BOOL)shouldBecomeFirstResponder view:(UIView *)view
-{
-    if (shouldBecomeFirstResponder) {
-        [view becomeFirstResponder];
-    } else {
-        [view resignFirstResponder];
-    }
-}
-
-+ (BOOL)commonTextUIValidate:(BOOL)isRequired hasText:(BOOL)hasText predicate:(NSPredicate *)predicate text:(NSString *)text error:(NSError *__autoreleasing *)error
+- (void)addObserverWithCompletion:(CompletionHandler)completion
 {
-    if (isRequired && !hasText) {
-        if (error) {
-            *error = [NSError errorWithDomain:ACRInputErrorDomain code:ACRInputErrorValueMissing userInfo:nil];
-        }
-        return NO;
-    } else if (hasText && predicate) {
-        return [predicate evaluateWithObject:text];
-    }
-    return YES;
+    NSObject<ACRIBaseInputHandler> *inputHandler = [self getInputHandler];
+    [inputHandler addObserverWithCompletion:completion];
 }
 
-// returns intrinsic content size for inputs
-- (CGSize)intrinsicContentSize
+- (void)resetInput
 {
-    CGFloat width = 0.0f, height = 0.0f;
-    for (UIView *view in self.stack.arrangedSubviews) {
-        if (!view.hidden) {
-            CGSize size = [view intrinsicContentSize];
-            width = MAX(size.width, width);
-            height += size.height;
-        }
-    }
-
-    return CGSizeMake(width, height);
+    [[self getInputHandler] resetInput];
```

Comment:
This view has input validation, should we reset the validation status when reset the input?

Reply:
since the validate function has side effect for rendering, added that